### PR TITLE
Data picker dropdown: arrays & objects

### PR DESCRIPTION
**Pull request by bkrmendy**
https://utopia.fish/p/c92053d9-cottony-decade/?branch_name=feature-insert-array-obejct

## Problem
The data picker popup cannot sensibly insert object props and array elements yet

## Fix 
Improve the data picker popup to insert object props and array elements.

### CODE REVIEW

File: editor/src/core/shared/javascript-cache.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export function resolveParamsAndRunJsCode(
     ...javascriptBlock,
     definedElsewhere: Object.keys(definedElsewhereInfo),
   }
+
```
Comment (ruggi) [inline editor/src/core/shared/javascript-cache.ts:37]: 👾

**Review by ruggi — APPROVED** [review]

File: editor/src/core/shared/javascript-cache.ts (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/core/shared/javascript-cache.ts:37]: 🔭

File: editor/src/components/inspector/sections/component-section/component-section.tsx [hunk 2]
```diff
@@ -1142,7 +1128,59 @@ export class ComponentSection extends React.Component<
...
+  if (Array.isArray(value)) {
+    return value.flatMap((v, idx) =>
+      valuesFromVariable(`${name}[${idx}]`, v).map((variable) =>
+        patchDefinedElsewhereInfo(variable),
+      ),
+    )
+  }
+
+  return Object.entries(value).flatMap(([key, field]) =>
+    valuesFromVariable(`${name}.${key}`, field).map((variable) =>
```
Comment (balazsbajorics) [inline editor/src/components/inspector/sections/component-section/component-section.tsx:1161]: beware of objects like this `{['I am legal JS']: 'beware of me'}`, I _think_ this code would produce `objectName.I am legal JS` as the code

Comment (bkrmendy) [reply editor/src/components/inspector/sections/component-section/component-section.tsx:1161]: I'll refactor it to `objectName['I am legal JS']`

**Review by gbalint — APPROVED** [review]

File: editor/src/components/inspector/sections/component-section/component-section.tsx (hunk 2 above)
Comment (bkrmendy) [reply editor/src/components/inspector/sections/component-section/component-section.tsx]: https://github.com/concrete-utopia/utopia/pull/4858/commits/602f3c9c6c3c97b30d43d27be1b59c95983c1f74

**Review by seanparsons — APPROVED** [review]

File: editor/src/core/shared/javascript-cache.ts (hunk 1 above)
Comment (bkrmendy) [reply editor/src/core/shared/javascript-cache.ts]: https://github.com/concrete-utopia/utopia/pull/4858/commits/f3c89f172776a95199ca1280fa24519567b979b5

**Pull request merged**